### ISSUE TITLE
fine-tuning RuBERT | data format for Text Classification Task

### ISSUE THREAD

**Issue by vcjob** [deepmipt/DeepPavlov#1079]
Hello everyone!

I have the following problem: I've got a dataset (currently - very small, ~550 documents. In the future - around 10k documents) and I need to do a multi-classification task on it, moreover, each document may have multiple class labels attached to it. 
I've read the paper https://blog.insightdatascience.com/using-transfer-learning-for-nlp-with-small-data-71e10baf99a6 , which states that even with 500 samples there is quite a gain in BERT accuracy. So I want to give BERT a try. I've tried to use it's embedding as an input to RandomForest, and the results were worse than with tf-idf. So I hope that having BERT fine-tuned on the target domain documents will improve it's accuracy up to overcoming the RandomForest on tf-idf (with 10k dataset)/

I've red the issue [#1065](https://github.com/deepmipt/DeepPavlov/issues/1065) and understood I need to follow google's guide and use DeepPavlov's ruBERT and vocab as well. But the question I have is, what should the dataset look like?
I need to do classification. I suppose I may use just BERT's embedding, and do using it a one-verse-the-rest classification with other classifiers. So, should I just fine-tune the BERT somehow in unsupervised manner, like providing the plain text? Or I must provide the class labels as well? In that case, having like 10 classes, while most of the documents are sharing multiple classes each, what would be the right way to fine-tune BERT to obtain better embeddings? Is that possible after all? Or the only way is to tine-tune multiple BERT models (which is .. insane).

Thank you a lot for any suggestions!

**Comment by dilyararimovna** [comment]
Hello!

It's probably a better decision to choose not the way of using only embeddings from BERT but to use a BERT-based classification model itself.
[Here](https://github.com/deepmipt/DeepPavlov/blob/master/deeppavlov/configs/classifiers/rusentiment_convers_bert.json) you may find a config file - example of how to use pre-trained BERT-based model for text classification. Take into account, that in this example conversational ruBERT is being used. So, if you have more official text style, you are better to use pre-trained ruBERT ([here](https://github.com/deepmipt/DeepPavlov/blob/master/deeppavlov/configs/classifiers/paraphraser_rubert.json) an example of using ruBERT but for classification of pairs of texts. So, you can take paths to pre-trained ruBERT from this config).

As an important difference from the first config example, you should set parameter `multilabel` for `bert_classifier` to `True` (see [here](https://github.com/deepmipt/DeepPavlov/blob/master/deeppavlov/models/bert/bert_classifier.py)).
Also, in this case you need to change
```
      {
        "in": "y_pred_probas",
        "out": "y_pred_ids",
        "class_name": "proba2labels",
        "max_proba": true
      },
```
to 
```
      {
        "in": "y_pred_probas",
        "out": "y_pred_ids",
        "class_name": "proba2labels",
        "confident_threshold": 0.5
      },
```

Description of dataset readers and how the data should be formatted you may find [here](https://github.com/deepmipt/DeepPavlov/blob/master/examples/classification_tutorial.ipynb).

Feel free to ask questions!
